ibu-imager/clusterinfo/clusterinfo.go

```diff
@@ -54,18 +57,40 @@ func (m *InfoClient) CreateClusterInfo(ctx context.Context) (*ClusterInfo, error
 	if err := m.client.Get(ctx, types.NamespacedName{Name: "version"}, clusterVersion); err != nil {
 		return nil, err
 	}
+
 	installConfig, err := m.getInstallConfig(ctx)
 	if err != nil {
 		return nil, err
 	}
+
+	ip, err := m.getNodeInternalIP(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	return &ClusterInfo{
 		ClusterName: installConfig.Metadata.Name,
 		Domain:      installConfig.BaseDomain,
 		Version:     clusterVersion.Status.Desired.Version,
 		ClusterID:   string(clusterVersion.Spec.ClusterID),
+		MasterIP:    ip,
 	}, nil
 }
 
+func (m *InfoClient) getNodeInternalIP(ctx context.Context) (string, error) {
+	node, err := utils.GetSNOMasterNode(ctx, m.client)
+	if err != nil {
+		return "", err
+	}
+	for _, addr := range node.Status.Addresses {
```

Comment:
what about dualstack? if we don't support let's error on len(node.Status.Addresses)

Reply:
We will support, just need to have such system up and running and i will handle it in next prs

Reply:
so for now let's error

Reply:
This will require going through all the Ips and getting the list and checking that list has more than one value. Not sure it is worth now. I will open ticket for dual stuck. Currently this code is only for our internal use so it should be fine.
I will add TODO for it though